src/semgrep_agent/semgrep_app.py

```diff
@@ -104,13 +106,15 @@ def fetch_rules_text(self) -> str:
         else:
             return response.text
 
-    def download_rules(self) -> Path:
+    def download_rules(self) -> Tuple[Path, int]:
         """Save the rules configured on semgrep app to a temporary file"""
         # hey, it's just a tiny YAML file in CI, we'll survive without cleanup
         rules_file = tempfile.NamedTemporaryFile(suffix=".yml", delete=False)  # nosem
         rules_path = Path(rules_file.name)
-        rules_path.write_text(self.fetch_rules_text())
-        return rules_path
+        rules = self.fetch_rules_text()
+        parsed = yaml.load(rules, Loader=yaml.Loader)
```

Comment:
I believe we use a different loader on the backend, the RT (or round-trip) loader. Is this intentionally different?

Reply:
hmm I didn't see that and am still not finding it. Do you have a file name and line number I can look at? 

I tried using `yaml.load(...)` with no loader first, but it told me that was insecure and that I should use `Loader=yaml.Loader` unless that did not work for my purposes 🤷‍♀️

Reply:
Oh, ok, just read https://yaml.readthedocs.io/en/latest/api.html and found line 36 in util.py in backend code. I'll update this